test/runtests.jl

```diff
@@ -1,5 +1,35 @@
 using TiledIteration, OffsetArrays
 using Test
+using Documenter
+
+if VERSION < v"1.6-"
+    Documenter.doctest(TiledIteration)
+    # Version restriction can be lifted, when
+    # filters can be passed to `doctest`
+    # See https://github.com/JuliaDocs/Documenter.jl/pull/1435
+    #
+    # doctestfilters = [
+    #     r"{([a-zA-Z0-9]+,\s?)+[a-zA-Z0-9]+}",
+    #     r"(Array{[a-zA-Z0-9]+,\s?1}|Vector{[a-zA-Z0-9]+})",
+    #     r"(Array{[a-zA-Z0-9]+,\s?2}|Matrix{[a-zA-Z0-9]+})",
+    # ]
+    # Documenter.doctest(TiledIteration, doctestfilters = doctestfilters)
+end
+
+
+@testset "TileIterator small examples" begin
+    titr = @inferred TileIterator((1:10,), RelaxLastTile((3,)))
+    @test titr == [(1:3,), (4:6,), (7:9,), (10:10,)]
+
+    titr = @inferred TileIterator((1:10,), RelaxStride((3,)))
+    @test titr == [(1:3,), (3:5,), (6:8,), (8:10,)]
+
+    titr = @inferred TileIterator((1:4,), RelaxStride((2,)))
+    @test titr == [(1:2,), (3:4,)]
+
+    titr = @inferred TileIterator((1:4,), RelaxLastTile((2,)))
+    @test titr == [(1:2,), (3:4,)]
+end
```

Comment:
As a special case, it would be very helpful to add some tests for other range types, e.g., `OffsetArrays.IdOffsetRange` and `Base.IdentityUnitRange`; those are usually very good test cases to catch exceptions.

```julia
# unsure if this works
using OffsetArrays
A = zeros(10)
AO = OffsetArray(A, OffsetArrays.Origin(0))

titr = TileIterator(axes(AO), RelaxStride((3, )))
@test titr == [(0:2,), (3:5,), (6:8,), (9:9,)]
```

Reply:
done